nixos/modules/services/web-apps/immich.md

```diff
@@ -26,4 +26,238 @@ Using the following configuration you can setup a basic installation of Immich:
+that all photos, videos and thumbnails are going to appear with the
+path prefix `upload/` in the database dump, as Immich stores its data
+in `./upload` relative to its working directory by default.
+See [Immich "Files Custom Locations" documentation](https://immich.app/docs/guides/custom-locations/)
+for more information.
+In summary, the goal is to replace the `upload/` prefix with the new
+*absolute* media location path `TARGET_LOCATION`.
+
+In order to change these file paths in the database dump, we can use
+the following `sed` command:
+
+```console
+# sed "s#upload/(.*)/#$TARGET_LOCATION/\1/#g" ./immich-fixed-search-path.sql > ./immich-final.sql
+```
+
+This `sed` command makes use of capture groups as just replacing every
+occurrence of `upload/` would break paths referencing
+`upload/upload/`. In order to replace only the prefix and not all
+occurrences of `upload`, the left-hand-side also captures the path
+segment after `upload/`.
+
+### Prepare new database and library {#module-immich-migrating-from-docker-compose-prepare-db}
+
+To ensure that the new database does not contain any residual data
+from the brief time Immich was initially started on the system, we are going to
+drop all existing data. To do so, first start an interactive `psql`
+session as the Postgres superuser in the `immich` database. The
+superuser is needed to create some of the extensions later; the
+`immich` user will not be able to do that even in its own database.
+
+```console
+# sudo -u postgres psql immich
+psql (15.8)
+Type "help" for help.
+
+immich=#
+```
+
+Now enter the following SQL code to recreate the database schemas:
+
+::: {.warning}
+**THIS IS GOING TO DELETE ALL DATA IN THE IMMICH DATABASE**. Ensure
+that you are logged into the right database and that no valuable data
+is going to be deleted in the `public` and `vectors` schemas.
+:::
+
+<!-- Ensure this snippet is up to date with the module. -->
+<!-- TODO: Provide a migration script that utilizes the init script generated by the module? -->
+```sql
+-- Drop existing schemas
+DROP SCHEMA public CASCADE;
+DROP SCHEMA vectors CASCADE;
+
+-- Create schemas and extensions
+CREATE SCHEMA public;
+
+CREATE EXTENSION IF NOT EXISTS unaccent;
+CREATE EXTENSION IF NOT EXISTS "uuid-ossp";
+CREATE EXTENSION IF NOT EXISTS vectors;
+CREATE EXTENSION IF NOT EXISTS cube;
+CREATE EXTENSION IF NOT EXISTS earthdistance;
+CREATE EXTENSION IF NOT EXISTS pg_trgm;
+
+-- Grant access to immich user
+ALTER SCHEMA public OWNER TO immich;
+ALTER SCHEMA vectors OWNER TO immich;
+GRANT SELECT ON TABLE pg_vector_index_stat TO immich;
+
+-- Optionally "update" extension (this is technically a noop)
+ALTER EXTENSION vectors UPDATE;
+```
+
+Once the above statements have been executed, you may exit the
+interactive `psql` console by entering `\q` or pressing `Ctrl + D`.
+
+Now we can import the modified database dump using the following
+command:
+
+```console
+# sudo -u postgres psql immich < ./immich-final.sql
```

Comment:
If run like this, the immich user will not have any privileges on the new database objects. I had to run

```sql
GRANT ALL PRIVILEGES ON ALL TABLES IN SCHEMA public TO immich;
GRANT ALL PRIVILEGES ON ALL TABLES IN SCHEMA vectors TO immich;
GRANT USAGE, SELECT ON ALL SEQUENCES IN SCHEMA public TO immich;
GRANT USAGE, SELECT ON ALL SEQUENCES IN SCHEMA vectors TO immich;
```

after the import.

Reply:
Hmm, interesting. I don't think this was needed in my case. But I guess it wouldn't hurt doing this just in case.